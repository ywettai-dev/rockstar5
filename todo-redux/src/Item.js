import React from 'react';
import { connect } from 'react-redux';

class Item extends React.Component {
   render() {
      return (
         <li>
            <input
               type="checkbox"
               checked={this.props.item.status}
               onChange={() => {
                  this.props.toggle(this.props.item._id)
               }}
            >
            </input>
            {this.props.item.subject}
            <a href="#/" onClick={() => {
               this.props.delete(this.props.item._id)
            }}>&times;</a>
         </li>
      )
   }
}

const NewItem = connect(null, dispatch => {
   return {
      delete: _id => dispatch({ type: 'DELETE', _id: _id }),
      toggle: _id => dispatch({ type: 'TOGGLE', _id: _id })
   }
})(Item);

export default NewItem;