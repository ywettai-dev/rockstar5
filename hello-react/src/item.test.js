import React from 'react';
import Item from './item';
import renderer from 'react-test-renderer';

test('Item Component', () => {
   const item = { '_id': 1, 'subject': 'Sample' }
   const snapshot = renderer.create(<Item item={item} />).toJSON();

   expect(snapshot).toMatchSnapshot();
});