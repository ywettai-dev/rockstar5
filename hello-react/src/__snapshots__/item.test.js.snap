// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component 1`] = `
<div>
  <li>
    <a
      href="#/"
      onClick={[Function]}
    >
      Ã—
    </a>
  </li>
</div>
`;
